.buildkite/pipeline.yml

```diff
@@ -0,0 +1,38 @@
+# SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+steps:
+  - label: REUSE lint
+    commands:
+      - nix-build -A checks.x86_64-linux.reuse-lint
+
+  - label: Check trailing whitespaces
+    commands:
+      - nix-build -A checks.x86_64-linux.trailing-whitespace
+
+  - label: files generated by stack2cabal are up to date
+    commands:
+      - nix-shell --run "stack2cabal"
+      - git diff --exit-code
+
+  - label: build
+    commands:
+      - nix-build -A checks.x86_64-linux.build-all
+
+  - label: hlint
+    commands:
+      - nix run -f. legacyPackages.x86_64-linux.haskellPackages.hlint
+        -c hlint -j --hint .hlint.yaml */
+
+  - label: weeder
+    commands:
+      - nix-build -A weeder-script
+      - ./result
+
+  # don't run tests until all components are built
```

Comment:
No real reason to do this, nix can handle building&running tests together with the rest of the checks